"failed to parse rustc JSON"

I just got this while working on Miri tests:
```
The application panicked (crashed).
Message:  failed to parse rustc JSON output at line 0: invalid value: integer `0`, expected a nonzero usize at line 1 column 755: {"$message_type":"diagnostic","message":"`main` function not found in crate `thread_panic`","code":{"code":"E0601","explanation":"No `main` function was found in a binary crate.\n\nTo fix this error, add a `main` function:\n\n```\nfn main() {\n    // Your program will start here.\n    println!(\"Hello world!\");\n}\n```\n\nIf you don't know the basics of Rust, you can look at the\n[Rust Book][rust-book] to get started.\n\n[rust-book]: https://doc.rust-lang.org/book/\n"},"level":"error","spans":[{"file_name":"tests/pass/panic/thread_panic.rs","byte_start":0,"byte_end":0,"line_start":0,"line_end":0,"column_start":1,"column_end":1,"is_primary":true,"text":[],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"consider adding a `main` function to `tests/pass/panic/thread_panic.rs`","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"error[E0601]: `main` function not found in crate `thread_panic`\n   |\n   = note: consider adding a `main` function to `tests/pass/panic/thread_panic.rs`\n\n"}

Location: /home/r/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ui_test-0.21.2/src/rustc_stderr.rs:185
```
Looks like something is making an assumption that some integer must be non-zero when actually it can be zero?

--- a/src/rustc_stderr.rs
+++ b/src/rustc_stderr.rs
@@ -67,12 +67,17 @@ fn span_line(span: &DiagnosticSpan, file: &Path, primary: bool) -> Option<spanne
             };
         }
     }
-    ((!primary || span.is_primary) && file_name == file).then_some(spanned::Span {
-        file: file_name,
-        line_start: span.line_start.try_into().unwrap(),
-        line_end: span.line_end.try_into().unwrap(),
-        col_start: span.column_start.try_into().unwrap(),
-        col_end: span.column_end.try_into().unwrap(),
+    ((!primary || span.is_primary) && file_name == file).then(|| {
+        let span = || {
+            Some(spanned::Span {
+                file: file_name,
+                line_start: span.line_start.try_into().ok()?,
+                line_end: span.line_end.try_into().ok()?,
+                col_start: span.column_start.try_into().ok()?,
+                col_end: span.column_end.try_into().ok()?,
+            })
+        };
+        span().unwrap_or_default()
     })
 }
 

